From 05b16745a1d26c4e098c39a49c56e9c312771bd2 Mon Sep 17 00:00:00 2001
From: Ashish SHUKLA <ashish.is@lostca.se>
Date: Mon, 23 Sep 2024 12:43:37 +0200
Subject: [PATCH] Fix release build with GCC 11.x
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

meson in release build mode (with GCC) triggers following compiler error:

   In file included from ../source/clipboard.c:14:
   ../source/clipboard.c: In function ‘select_mime_type_for_offer’:
   ../source/log.h:58:5: error: ‘%s’ directive argument is null [-Werror=format-overflow=]
      58 |     log_msg(LOG_CLASS_DEBUG, LOG_MODULE, __FILE__, __LINE__, fmt, ## __VA_ARGS__)
         |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ../source/clipboard.c:81:5: note: in expansion of macro ‘LOG_DBG’
      81 |     LOG_DBG("mime-type: %s -> %s (offered type was %s)",
         |     ^~~~~~~
   ../source/clipboard.c:81:31: note: format string is defined here
      81 |     LOG_DBG("mime-type: %s -> %s (offered type was %s)",
         |                               ^~

Fixes #420
---
 clipboard.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/clipboard.c b/clipboard.c
index a4fa2d2..c692503 100644
--- a/clipboard.c
+++ b/clipboard.c
@@ -79,7 +79,9 @@ select_mime_type_for_offer(const char *_mime_type,
     }
 
     LOG_DBG("mime-type: %s -> %s (offered type was %s)",
-            mime_type_map[*type], mime_type_map[mime_type], _mime_type);
+            mime_type_map[*type],
+            (mime_type_map[mime_type] == NULL ? "(null)" : mime_type_map[mime_type]),
+            _mime_type);
 
     /* Mime-type transition; if the new mime-type is "better" than
      * previously offered types, use the new type */
